core/ROB.cpp

```diff
@@ -198,6 +204,7 @@ namespace olympia
 
     void ROB::onStartingTeardown_() {
         if ((reorder_buffer_.size() > 0) && (false == rob_stopped_simulation_)) {
+            rob_drained_notif_source_->postNotification(true);
```

Comment:
Based on the discussion, isn't this supposed to be set on the else branch?

> When Retire is completely drained AND is stopping simulation, that notification can be broadcasted, which LS can "hear" and set a boolean retire_done_and_is_drained_ and then you can check the ensure the LS issue queue is actually drained.

https://github.com/riscv-software-src/riscv-perf-model/pull/90#issuecomment-1739405558

Reply:
☝🏻 good question (and good catch)

I think it's better if that notification is posted _at the point_ the ROB stops simulation on the scheduler.  And the LS should implement the `onStartingTeardown_` and check for this condition there and not the destructor.  That'd be pretty clean and ordering will be consistent.

Reply:
@knute-sifive I couldnt find find a good place to send the notification. The rob stops the scheduler only when ilimit parameter is passed to olympia but what about the case when it is stopped naturally

Reply:
> Based on the discussion, isn't this supposed to be set on the else branch?
> 
> > When Retire is completely drained AND is stopping simulation, that notification can be broadcasted, which LS can "hear" and set a boolean retire_done_and_is_drained_ and then you can check the ensure the LS issue queue is actually drained.
> 
> [#90 (comment)](https://github.com/riscv-software-src/riscv-perf-model/pull/90#issuecomment-1739405558)

Good point, I updated the code.